OCPP-J/src/main/java/eu/chargetime/ocpp/WebSocketListener.java

```diff
@@ -122,19 +124,25 @@ public ServerHandshakeBuilder onWebsocketHandshakeReceivedAsServer(WebSocket web
                             .InternetAddress(webSocket.getRemoteSocketAddress())
                             .build();
 
-            String username = null, password = null;
+            byte[] username = null, password = null;
             if (clientHandshake.hasFieldValue("Authorization")) {
               String authorization = clientHandshake.getFieldValue("Authorization");
               if (authorization != null && authorization.toLowerCase().startsWith("basic")) {
                 // Authorization: Basic base64credentials
                 String base64Credentials = authorization.substring("Basic".length()).trim();
                 byte[] credDecoded = Base64.getDecoder().decode(base64Credentials);
-                String credentials = new String(credDecoded, StandardCharsets.UTF_8);
-                // credentials = username:password
-                final String[] values = credentials.split(":", 2);
-                if (values.length >= 1) username = values[0];
-                if (values.length >= 2) password = values[1];
+                // split credentials on username and password
+                for (int i = 0; i < credDecoded.length; i++) {
+                  if (credDecoded[i] == ':') {
+                    username = Arrays.copyOfRange(credDecoded, 0, i);
+                    if (i != credDecoded.length - 1) {
```

Comment:
I would prefer (i + 1 < credDecoded.length) as the condition to be on the safe side.

Reply:
Done, please review

Reply:
Thanks. Looks all good to me now.